framework/decode/openxr_replay_consumer_base.cpp

```diff
@@ -843,17 +865,40 @@ void OpenXrReplayConsumerBase::Process_xrPollEvent(const ApiCallInfo&
         eventData->IsNull() ? nullptr : eventData->AllocateOutputData(1, { XR_TYPE_EVENT_DATA_BUFFER, nullptr });
     InitializeOutputStructNext(eventData);
 
-    XrResult replay_result;
+    // If we have not previously started tracking the event behavior for this event type, start tracking it.
+    if (event_behavior_tracking_.find(capture_event->type) == event_behavior_tracking_.end())
+    {
+        event_behavior_tracking_[capture_event->type] = EventBehaviorTracking();
```

Comment:
`std::unordered_map::operator[]` automatically inserts a default initialized entry, when key is not found.  So this block is unneeded.

What would be useful instead is a reference `EventBehaviorTracking &behavior_tracking = event_behavior_tracking_[capture_event->type]` given that '[]' is best O(1) is worst case O(n).  and that same dereference is done many times below.

Reply:
I didn't know that about the initialize.  Thanks.